services/meta/service.go

```diff
@@ -108,21 +108,25 @@ func (s *Service) serve() {
 // Close closes the underlying listener.
 func (s *Service) Close() error {
 	if s.ln != nil {
-		return s.ln.Close()
-	}
-
-	if err := s.store.close(); err != nil {
-		return err
+		if err := s.ln.Close(); err != nil {
+			return err
+		}
 	}
+	s.handler.Close()
```

Comment:
Right, but `handler.close()` doesn't wait to clean up either, it just exits.  There could be valid code running (like a snapshot, etc.).  If you stomp on the underlying store before those finish I'm not sure what happens.  The code may already handle this, I haven't looked through all the code paths.  I was more asking that if we do need to wait, that there is no blocker in the handler code to stop exiting if a connection is actually sending data back.